src/openfermion/utils/_trotter_qasm_conversion.py

```diff
@@ -0,0 +1,331 @@
+
+	elif trotter_order == 2:
+		if len(term_ordering) < 2:
+			raise ValueError("Not enough terms in the Hamiltonian to do "+\
+				"second order trotterization")
+		for op in term_ordering[:-1]:
+			ret_val.append(QubitOperator(op,hamiltonian.terms[op]) * k_exp 
+				/ (2.0 * trotter_steps))
+		ret_val.append(QubitOperator(term_ordering[-1],
+			hamiltonian.terms[term_ordering[-1]] * k_exp / trotter_steps))
+
+		for op in reversed(term_ordering[:-1]):
+			ret_val.append(QubitOperator(op,hamiltonian.terms[op]) * k_exp 
+				/ (2.0 * trotter_steps))
+
+		for step in range(trotter_steps - 1):
+			ret_val.extend(ret_val)
+
+	elif trotter_order == 3:
+		if len(term_ordering) < 2:
+			raise ValueError("Not enough terms in the Hamiltonian to do "+\
+				"third order trotterization")
+		hamiltonian /= trotter_steps
+		hamiltonian *= k_exp
+		atemp = QubitOperator(term_ordering[0],hamiltonian.terms[term_ordering[0]])
+		btemp = term_ordering[1:]
+		third_order_trotter_helper(hamiltonian.terms, ret_val,atemp,btemp)
+
+		for step in range(trotter_steps - 1):
+			ret_val.extend(ret_val)
+	return (ret_val,num_qubits)
+
+
+
+def print_qubit_op_to_qasm(ostrm,qubop):
+	'''
+	This function exponentiates a single-Pauli-string QubitOperator
+	and prints it in QASM format to a supplied output stream
+
+	Params:
+		ostrm: file output stream
+		qubop: single Pauli-term QubitOperator to be printed
+		kexp (optional): float to be multiplied to the coefficient 
+	
+	Returns:
+		None
+	'''
+
+
+	for term in qubop.terms:
+
+		termCoeff = qubop.terms[term]
+
+		# List of operators and list of qubit ids
+		ops = list()
+		qids = list()
+		strBas1 = ""  # Basis rotations 1
+		strBas2 = ""  # Basis rotations 2
+		nl = "\n"  # New line
+
+		for p in term:	# p = single pauli term
+			qid = p[0]	# Qubit index
+			pop = p[1]	# Pauli op
+
+			qids.append(qid)  # Qubit index
+			ops.append(pop)	 # Pauli operator
+
+			if pop == 'X':
+				strBas1 += "H " + str(qid) + nl	 # Hadamard
+				strBas2 += "H " + str(qid) + nl	 # Hadamard
+			elif pop == 'Y':
+				strBas1 += "Rx 1.57079632679 " + str(qid) + nl # Z --> Y
+				strBas2 += "Rx -1.57079632679 " + str(qid) + nl	 # Y --> Z
+
+		# Prep for CNOTs
+		cnotPairs = np.vstack((qids[:-1], qids[1:]))
+		cnots1 = ""
+		cnots2 = ""
+		for i in range(cnotPairs.shape[1]):
+			pair = cnotPairs[:, i]
+			cnots1 += "CNOT " + str(pair[0]) + " " + str(pair[1]) + nl
+		for i in np.arange(cnotPairs.shape[1])[::-1]:
+			pair = cnotPairs[:, i]
+			cnots2 += "CNOT " + str(pair[0]) + " " + str(pair[1]) + nl
+
+		# Exponentiating each Pauli string requires five parts
+
+		# 1. Perform basis rotations
+		ostrm.write(strBas1)
+
+		# 2. First set CNOTs
+		# Store in string, add later
+		ostrm.write(cnots1)
+
+		# 3. Rotation (Note kexp & Ntrot)
+		ostrm.write("Rz " + str(termCoeff) + " "+ str(qids[-1])+ nl)
+
+		# 4. Second set of CNOTs
+		ostrm.write(cnots2)
+
+		# 5. Rotate back to Z basis
+		ostrm.write(strBas2)
+
+
+def write_qubit_ops(op_list,ostrm):
```

Comment:
Try to stay consistent in parameter ordering, i.e. op_list, output_stream vs output_stream, op_list between the two functions to avoid bugs for other people calling these routines